### ISSUE TITLE
<img> tag in compiled helper gives "Unexpected token <"

### ISSUE THREAD

**Issue by mpwalkerdine** [kirbysayshi/vash#67]
Firstly thanks for the awesome template engine...

This works:
```bash
$ echo "vash.helpers.img = function(){<i></i>}" | vash --helper
vash.link( function anonymous(
/**/) {
try { 
var __vbuffer = this.buffer; 
var model = this.model; 
var html = this; 
html.vl = 0, html.vc = 0; 
html.vl = 0, html.vc = 0; 
{html.vl = 1, html.vc = 2; 

__vbuffer.push('<');

__vbuffer.push('i');

__vbuffer.push('');

__vbuffer.push('>');
html.vl = 1, html.vc = 4; 
html.vl = 1, html.vc = 5; 

__vbuffer.push('</');

__vbuffer.push('i');

__vbuffer.push('>');
html.vl = 1, html.vc = 8; 
}html.vl = 0, html.vc = 0; 
html.vl = 0, html.vc = 0; 
} catch( e ){ 
  html.reportError( e, html.vl, html.vc, "@{<i></i>}", "!LB!", true ); 
} 

}, {"simple":false,"modelName":"model","helpersName":"html","args":[""],"asHelper":"img"} )

```

But this doesn't (notice the stray `</img>` which looks unprocessed...):
```bash
$ echo "vash.helpers.img = function(){<img></img>}" | vash --helper
/home/vagrant/npm-global/lib/node_modules/vash/runtime.js:268
  throw e;
        ^
SyntaxError: Problem while compiling template at line 0, character 0.
Original message: Unexpected token <.
Context: 

     1 | try { 
     2 | var __vbuffer = this.buffer; 
     3 | var model = this.model; 
     4 | var html = this; 
     5 | html.vl = 0, html.vc = 0; 
     6 | html.vl = 0, html.vc = 0; 
     7 | {html.vl = 1, html.vc = 2; 
     8 | 
     9 | __vbuffer.push('<');
    10 | 
    11 | __vbuffer.push('img');
    12 | 
    13 | __vbuffer.push('');
    14 | 
    15 | __vbuffer.push('>');
    16 | html.vl = 1, html.vc = 6; 
    17 | </img>}html.vl = 0, html.vc = 0; 
    18 | html.vl = 0, html.vc = 0; 
    19 | } catch( e ){ 
    20 |   html.reportError( e, html.vl, html.vc, "@{<img></img>}", "!LB!", true ); 
    21 | } 


    at Function (native)
    at Object.runtime.link (/home/vagrant/npm-global/lib/node_modules/vash/runtime.js:308:26)
    at exports.compile (/home/vagrant/npm-global/lib/node_modules/vash/index.js:73:21)
    at compileSingleHelper (/home/vagrant/npm-global/lib/node_modules/vash/lib/helperbatch.js:113:10)
    at /home/vagrant/npm-global/lib/node_modules/vash/lib/helperbatch.js:27:11
    at Array.forEach (native)
    at compile (/home/vagrant/npm-global/lib/node_modules/vash/lib/helperbatch.js:25:23)
    at Socket.<anonymous> (/home/vagrant/npm-global/lib/node_modules/vash/bin/vash:81:63)
    at emitNone (events.js:72:20)
    at Socket.emit (events.js:163:7)

```

Obviously my actual templates have `alt` and `src` attributes on the image. The effect is the same. Am I missing something?

Thanks again,
Matt

**Comment by kirbysayshi** [comment]
I _think_ what's happening (have to confirm) is that vash has a list of markup tags it expects to be "voids", meaning they have no closing tag. If you remove the `</img>`, I'm guessing you won't have this problem, since that's valid HTML5.

Let me know if that fixes it! I'll try to think of a way to make the error message more descriptive if I can. And either way that closing `</img>` should be treated as markup.... so there's probably a bug in there regardless.

**Comment by mpwalkerdine** [comment]
:thumbsup: You were right on the money, using the empty form works fine.

Thanks!

**Comment by kirbysayshi** [comment]
I published v0.9.1, which should add more descriptive error messages in this case. Thanks again for the great bug report!
